sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -320,6 +304,12 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
         })
 
       lazy val numberOfWithdrawalRequestBoxes = newBoxes.count(box => box.isInstanceOf[WithdrawalRequestBox])
+      val backwardTransferLimitEnabled = stateStorage.getConsensusEpochNumber match {
```

Comment:
I think, we have an issue here.
Let's assumes we've just forged the first SC block after SCFork1 activation.
In this case during tx validation `backwardTransferLimitEnabled = false` because the `stateStorage.getConsensusEpochNumber` will return the actual tip epoch number that is before the fork 1.

It seems that we don't need to account the fork1 here at all. The only check is that `numberOfWithdrawalRequestBoxes > params.maxWBsAllowed`, which seems to be fork independent.
In other place we verify `checkWithdrawalBoxesAllowed` that seems enough.

Reply:
Seems, we can add this check to the semantic validity of `SidechainTransaction` to detect this case even earlier

Reply:
Good point.
Maybe it is harder to test it because at the moment we have 1000 output boxes limit per transaction but for sure it is the best point where to put it.